unlock-app/src/__tests__/middlewares/web3Middleware.test.js

```diff
@@ -378,6 +378,41 @@ describe('Lock middleware', () => {
     })
   })
 
+  describe('CREATE_LOCK', () => {
+    it('should generateLockAddress if the address is missing', () => {
+      expect.assertions(2)
+      const { next, invoke } = create()
+      const lock = {
+        name: 'my lock',
+      }
+      const address = '0x123'
+      const action = { type: CREATE_LOCK, lock }
+      mockWeb3Service.generateLockAddress = jest.fn(() => {
+        return Promise.resolve(address)
+      })
+
+      invoke(action)
+      expect(mockWeb3Service.generateLockAddress).toHaveBeenCalledWith()
```

Comment:
How do we feel about updating the assertion to `toHaveBeenCalled` as we aren't validating the passed value?